File: tests/tests_old_compiler/PajameterServer/pajatest.js
japp nspc {
    required_linker_flags: -lm;
}

jdata {
    struct gradient_update_t {
        int dataTag;
        unsigned long long int logicalId;
        int gradient[128];
    } gradients as uflow;
}


async function readGradients() {
    console.log("Reading the gradients...");
    while (1) {
        console.log("waiting for gradient updates...");
        var gradient_updates = await gradients.readLast();
        console.log("Got updates ..", gradient_updates);
    }
}

if (jsys.type === 'fog')
    readGradients();

File: tests/tests_old_compiler/PajameterServer/pajatest.c


jasync calculateGradient(int data[]) {
    int dataTag = 1;
    printf("computing gradient for data %d\n", dataTag);

    while (1) {
        printf("Before sleep... \n");
        jsys.sleep(10000000); // TODO
        jarray int gradient[128] = {1, 2, 3};

        printf("writing to uflow %d\n", dataTag);
        struct gradient_update_t gradient_wrapper = {.dataTag = dataTag, .logicalId = 1221, .gradient = gradient};
        gradients.write(&gradient_wrapper);
        dataTag++;
    }
}


int main(int argc, char* argv[]) {
    jarray int data[128] = {1, 2, 34, 5, 67, 76};

//    for (int i = 0; i < 1000; i++)
    calculateGradient(&data);

    return 0;
}
